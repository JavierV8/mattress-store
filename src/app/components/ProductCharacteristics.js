import React from 'react';
import PropTypes from 'prop-types';
import { Progress } from 'react-sweet-progress';
import 'react-sweet-progress/lib/style.css';

const ProductCharacteristics = props => {
    const { firmness, breathability, adaptability } = props;
    return (
        <div className="product-characteristics">
            <div className="product-characteristics-element">
                <h5>Firmness</h5>
                <Progress type="circle" width={70} percent={firmness} />
                <h5>intermediate</h5>
            </div>

            <div className="product-characteristics-element">
                <h5>Breathability</h5>
                <Progress type="circle" width={70} percent={breathability} />
                <h5>intermediate</h5>
            </div>

            <div className="product-characteristics-element">
                <h5>Adaptability</h5>
                <Progress type="circle" width={70} percent={adaptability} />
                <h5>intermediate</h5>
            </div>
        </div>
    );
};

ProductCharacteristics.propTypes = {
    firmness: PropTypes.number.isRequired,
    breathability: PropTypes.number.isRequired,
    adaptability: PropTypes.number.isRequired,
};

export default ProductCharacteristics;
