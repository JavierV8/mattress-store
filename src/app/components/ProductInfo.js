import React from 'react';
import ProductCharacteristics from '../components/ProductCharacteristics';

const ProductInfo = props => {
    const { name, image, firmness, breathability, adaptability, title, description, price } = props.productData;

    return (
        <div className="productInfo-container">
            <h1 className="product-name">{name}</h1>
            <div className="product-stars">	&#9733; 	&#9733; 	&#9733; 	&#9733; 	&#9733;</div>

            <div className="productInfo-body">

                <div className="productInfo-left">
                    <div className="productInfo-image">
                        <img
                            src={require("../static/images/" + image)}
                            alt={name}
                        />
                    </div>
                    <div className="productInfo-description">
                        <h1>Description</h1>
                        <ProductCharacteristics
                            firmness={firmness}
                            breathability={breathability}
                            adaptability={adaptability}
                        />
                        <h4 className="product-title">{title}</h4>
                        <p>{description}</p>
                    </div>
                </div>

                <div className="productInfo-right">
                    <p className="productInfo-right-price">{price} â‚¬</p>
                    <div>
                        <div className="productInfo-right-selectorBox">
                            <p className="productInfo-right-chosseTitle">Chosse size: </p>
                            <select className="productInfo-right-selector">
                                <option value="volvo">80X180</option>
                                <option value="saab">80X190</option>
                                <option value="opel">80X200</option>
                                <option value="audi">90X180</option>
                            </select>
                        </div>
                    </div>
                    <button className="productInfo-purchaseBox-addButton">ADD PRODUCT</button>
                </div>

            </div>
        </div>
    );
};

export default ProductInfo;