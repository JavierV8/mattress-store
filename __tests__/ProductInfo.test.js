import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import ProductInfo from '../src/app/components/ProductInfo';
import ProductCharacteristics from '../src/app/components/ProductCharacteristics';

Enzyme.configure({ adapter: new EnzymeAdapter() });

const setup = (props = {}) => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    return shallow(<ProductInfo {...props} />);
};

const findByTestAttr = (wrapper, val) => {
    return wrapper.find(`${val}`);
};

const props = {
    name: 'Mattress Chipre',
    image: 'Chipre.jpg',
    firmness: 7,
    breathability: 5,
    adaptability: 4,
    title: 'The best ally for the growth of your children',
    description: 'The Cyprus mattress stands out for its...',
    price: 44,
};

it('Should render ProductInfo without errors', () => {
    const wrapper = setup({
        productData: props,
    });
    const Component = findByTestAttr(wrapper, '.productInfo-container');
    expect(Component.length).toBe(1);
    expect(toJSON(wrapper)).toMatchSnapshot();
});

it('check prop name on h1 element', () => {
    const wrapper = setup({
        productData: props,
    });
    const Component = findByTestAttr(wrapper, '.product-name');
    expect(Component.text()).toEqual(props.name);
});

it('check prop title on h4 element', () => {
    const wrapper = setup({
        productData: props,
    });
    const Component = findByTestAttr(wrapper, '.product-title');
    expect(Component.text()).toEqual(props.title);
});

it('check prop description on p element', () => {
    const wrapper = setup({
        productData: props,
    });
    const Component = findByTestAttr(wrapper, '.productInfo-description');
    expect(Component.find('p').text()).toEqual(props.description);
});

it('check prop price on p element', () => {
    const wrapper = setup({
        productData: props,
    });
    const Component = findByTestAttr(wrapper, '.productInfo-right-price');
    expect(Component.text()).toEqual(`${props.price} â‚¬`);
});

it('Should render an image', () => {
    const wrapper = setup({
        productData: props,
    });
    const Component = findByTestAttr(wrapper, '.productInfo-image');
    expect(Component.find('img').length).toEqual(1);
});

it('Should render child component (ProductCharacteristics) whit props ', () => {
    const wrapper = setup({
        productData: props,
    });
    const Component = findByTestAttr(wrapper, '.productInfo-container');
    expect(Component.find(ProductCharacteristics).length).toBe(1);
    expect(Component.find('#childId').exists()).toBeFalsy();
    expect(Component.find(ProductCharacteristics).props().firmness).toBe(
        props.firmness
    );
    expect(Component.find(ProductCharacteristics).props().breathability).toBe(
        props.breathability
    );
    expect(Component.find(ProductCharacteristics).props().adaptability).toBe(
        props.adaptability
    );
});
