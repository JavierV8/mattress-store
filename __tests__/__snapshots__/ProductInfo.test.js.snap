// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ProductInfo without errors 1`] = `
<div
  className="productInfo-container"
>
  <h1
    className="product-name"
  >
    Mattress Chipre
  </h1>
  <div
    className="product-stars"
  >
     
    ★ ★ ★ ★ ★
  </div>
  <div
    className="productInfo-body"
  >
    <div
      className="productInfo-left"
    >
      <div
        className="productInfo-image"
      >
        <img
          alt="Mattress Chipre"
          src={
            Object {
              "default": "",
            }
          }
        />
      </div>
      <div
        className="productInfo-description"
      >
        <h1>
          Description
        </h1>
        <ProductCharacteristics
          adaptability={4}
          breathability={5}
          firmness={7}
        />
        <h4
          className="product-title"
        >
          The best ally for the growth of your children
        </h4>
        <p>
          The Cyprus mattress stands out for its...
        </p>
      </div>
    </div>
    <div
      className="productInfo-right"
    >
      <p
        className="productInfo-right-price"
      >
        44
         €
      </p>
      <div>
        <div
          className="productInfo-right-selectorBox"
        >
          <p
            className="productInfo-right-chosseTitle"
          >
            Chosse size:
             
          </p>
          <select
            className="productInfo-right-selector"
          >
            <option
              value="volvo"
            >
              80X180
            </option>
            <option
              value="saab"
            >
              80X190
            </option>
            <option
              value="opel"
            >
              80X200
            </option>
            <option
              value="audi"
            >
              90X180
            </option>
          </select>
        </div>
      </div>
      <button
        className="productInfo-purchaseBox-addButton"
        type="button"
      >
        ADD PRODUCT
      </button>
    </div>
  </div>
</div>
`;
